import CredentialsProvider from "next-auth/providers/credentials"
import { NuxtAuthHandler } from "#auth"

export default NuxtAuthHandler({
  // a) Never hardcode your secret in your code!! and b) use a secure secret, `test-123` is **not** secure!!
  secret: process.env.AUTH_SECRET || "my-auth-secret",
  pages: {
    // Change the default behavior to use `/login` as the path for the sign-in page
    signIn: "/auth/signin",
  },
  // session: {
  //   strategy: 'jwt'
  // },
  providers: [
    // @ts-expect-error You need to use .default here for it to work during SSR. May be fixed via Vite at some point
    CredentialsProvider.default({
      // The name to display on the sign in form (e.g. 'Sign in with...')
      name: "Credentials",
      authorize(credentials: any) {
        // You need to provide your own logic here that takes the credentials
        // submitted and returns either a object representing a user or value
        // that is false/null if the credentials are invalid.
        // NOTE: THE BELOW LOGIC IS NOT SAFE OR PROPER FOR AUTHENTICATION!
        const user = { id: "1", name: "J Smith", username: "jsmith", password: "hunter2" }

        if (credentials?.username === user.username && credentials?.password === user.password) {
          // Any object returned will be saved in `user` property of the JWT
          return user
        } else {
          console.error("Warning: Malicious login attempt registered, bad credentials provided")
          // If you return null then an error will be displayed advising the user to check their details.
          return null
          // You can also Reject this callback with an Error thus the user will be sent to the error page with the error message as a query parameter
        }
      },
    }),
  ],
  callbacks: {
    // @ts-expect-error
    async jwt({ token, user }) {
      return { ...token, ...user }
    },
    // @ts-expect-error
    async session({ session, token, user }) {
      session.user = token as any
      return session
    },
  },
})
